ecs-agent/acs/session/task_manifest_responder_test.go

```diff
@@ -0,0 +1,186 @@
+//go:build unit
+// +build unit
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package session
+
+import (
+	"sync"
+	"testing"
+
+	"github.com/aws/amazon-ecs-agent/ecs-agent/acs/model/ecsacs"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/metrics"
+	mock_metrics "github.com/aws/amazon-ecs-agent/ecs-agent/metrics/mocks"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/wsclient"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	Cluster                                  = "test-cluster"
+	ContainerInstanceARN                     = "test-instance-arn"
+	MessageID                                = "test-messageID"
+	ManifestARN                              = "manifest-task-arn"
+	StartingLatestTaskManifestSequenceNumber = 15
+)
+
+// fakeContainerInstanceDirector imitates the ContainerInstanceDirector.
+type fakeContainerInstanceDirector struct {
```

Comment:
ecs-agent module is unaware of/doesn't have any container instance director. Could we generate and use mock implementations of the `SequenceNumberAccessor` and `TaskComparer` interfaces instead for this test file?